app/uk/gov/hmrc/incorporatedentityidentificationfrontend/connectors/CreateBusinessVerificationJourneyConnector.scala

```diff
@@ -57,18 +58,20 @@ class CreateBusinessVerificationJourneyConnector @Inject()(http: HttpClient,
 }
 
 object CreateBusinessVerificationJourneyConnector {
-  implicit object BusinessVerificationHttpReads extends HttpReads[Option[String]] {
-    override def read(method: String, url: String, response: HttpResponse): Option[String] = {
+  implicit object BusinessVerificationHttpReads extends HttpReads[Either[JourneyCreationFailure, JourneyCreated]] {
+    override def read(method: String, url: String, response: HttpResponse): Either[JourneyCreationFailure, JourneyCreated] = {
       response.status match {
         case CREATED =>
           (response.json \ "redirectUri").asOpt[String] match {
             case Some(redirectUri) =>
-              Some(redirectUri)
+              Right(JourneyCreated(redirectUri))
```

Comment:
I really like this, it's much clearer what the different scenarios are now.

Reply:
Nice!